### DISCUSSION TITLE
brew tells me to update curl even though it's already updated

### DISCUSSION THREAD

**Discussion by JuliaMathias** [Homebrew/discussions#3181] [Linux]
Hey, I've run into a problem while trying to upgrade gh and I don't know how to solve it. After running `brew update gh`, I got the following error message:

```bash
Error: Please update your system curl.
Minimum required version: 7.41.0
Your curl version:
Your curl executable:
==> Downloading https://ghcr.io/v2/homebrew/core/ncurses/manifests/6.3
/bin/curl: error while loading shared libraries: libldap_r-2.4.so.2: cannot open shared object file: No such file or directory
Error: curl: Failed to download resource "ncurses_bottle_manifest"
Download failed: https://ghcr.io/v2/homebrew/core/ncurses/manifests/6.3
Error: 'curl' must be installed and in your PATH!
```

but when I run `sudo apt upgrade curl` I get this. And you can see that my curl version is already above 7.41.0:

```bash
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
curl is already the newest version (7.74.0-1.3ubuntu2).
Calculating upgrade... Done
The following package was automatically installed and is no longer required:
  libfwupdplugin1
Use 'sudo apt autoremove' to remove it.
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
```

I'm using Ubuntu 21.10

**Comment by SMillerDev** [comment]
What do `brew config` and `brew doctor` say?

**Comment by JuliaMathias** [comment]
@SMillerDev 

brew config:

```bash
Error: Please update your system curl.
Minimum required version: 7.41.0
Your curl version: 
Your curl executable: 
HOMEBREW_VERSION: 3.4.3
ORIGIN: https://github.com/Homebrew/brew
HEAD: 600d84e56eda931c3dd3b6a9ee61233d9bf2b054
Last commit: 4 weeks ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: ab5040f62247ae337489653254bd817f33645006
Core tap last commit: 3 weeks ago
Core tap branch: master
HOMEBREW_PREFIX: /home/linuxbrew/.linuxbrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_CORE_GIT_REMOTE: https://github.com/Homebrew/homebrew-core
HOMEBREW_DISPLAY: :0
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 2.6.8 => /home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.8/bin/ruby
CPU: octa-core 64-bit skylake
Clang: N/A
Git: 2.32.0 => /bin/git
/bin/curl: error while loading shared libraries: libldap_r-2.4.so.2: cannot open shared object file: No such file or directory
Curl: N/A
Kernel: Linux 5.13.0-39-generic x86_64 GNU/Linux
OS: Ubuntu 21.10 (impish)
Host glibc: 2.34
/usr/bin/gcc: 11.2.0
/usr/bin/ruby: N/A
glibc: N/A
gcc@5: N/A
xorg: N/A
```

brew doctor:

```bash
Error: Please update your system curl.
Minimum required version: 7.41.0
Your curl version: 
Your curl executable: 
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Unbrewed header files were found in /usr/local/include.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae, and may need to be deleted.

Unexpected header files:
  /usr/local/include/wkhtmltox/image.h
  /usr/local/include/wkhtmltox/pdf.h

Warning: Some installed formulae are not readable:
  eks-login: Calling bottle :unneeded is disabled! There is no replacement.
Please report this issue to the luizm/tap tap (not Homebrew/brew or Homebrew/core):
  /home/linuxbrew/.linuxbrew/Homebrew/Library/Taps/luizm/homebrew-tap/Formula/eks-login.rb:6
```

**Reply by SMillerDev** [reply]
> /bin/curl: error while loading shared libraries: libldap_r-2.4.so.2: cannot open shared object file: No such file or directory

It seems that curl is broken

**Reply by danielnachun** [reply]
Given that this is the system `curl`, it should be entirely unaffected by Homebrew.  You probably want to use your host package manager to reinstall both `curl` and `libldap` as either `curl` was not updated after a `libldap` or `libldap` was not updated after a `curl` update.

**Reply by JuliaMathias** [reply]
Do you guys have any idea how to fix this? I uninstalled curl, installed libldap_r-2.4.so.2 (my system only had the 2.5 version installed) and tried re-installing curl. curl was still not working so I uninstalled again and installed the snap version of curl. Now curl is working when I use it on its own but brew apparently can still not find it :/ @SMillerDev @danielnachun

**Comment by brunokim** [comment]
For anyone finding this thread in the future, I've faced a similar issue because I've added `./bin` to the PATH. Prefer using an absolute path instead like `/home/${USER}/bin`.

At some point the brew install script checks for the curl tool location, and has `PWD=/usr`. The PATH entry makes the system curl match with `./bin/curl`, and since this is different than `/usr/bin/curl`, the variable is stored in `HOMEBREW_CURL_PATH`. Later, when the script chdirs to `/home/linuxbrew/.linuxbrew/Homebrew`, the relative path `./bin/curl` no longer works.

For Homebrew developers: is it possible to always store absolute paths instead of relative paths for these variables?
